pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -457,8 +460,30 @@ func (pc *PodController) configurePodSecondaryNetwork(pod *corev1.Pod, networkLi
 			}
 		}
 
+		// Generate a new MAC address, if the secondary interface MAC was not provided in the
+		// Pod annotation.
+		var mac net.HardwareAddr
```

Comment:
would it be simple to add unit tests for this new logic (valid case + invalid cases)?